### DISCUSSION TITLE
Extending the null-coalescing operator

### DISCUSSION THREAD

**Discussion by ActivistInvestor** [dotnet/csharplang#8669] [Language Ideas]
There may be good reasons why this may not be feasible, but I would find it helpful if the null-coalescing operator were extensible. 

So for example, Instead of

`int x = y != null ? 1 : 0;`

it could use ?? thusly:

`int x = y ?? 1 : 0;`

What that effectively means is that `y ??` is shorthand for `y != null`, and where the result and right-side operand of ?? is inferred by the context and/or the types on the right side.

**Comment by CyrusNajmabadi** [comment]
this becomes *heavily* syntactically ambiguous.  For example, this is already legal today:  `a ? b ?? 1 : 0`  Now the latter would become part of the `??` and you'd have different meaning between that *and* `a ? b ?? 1 : 0 : -1`.  Not a good space syntactically.

**Reply by ActivistInvestor** [reply]
I suppose it depends on operator precedence, and I realized there could be ambiguity within nested operations, and that's why I asked about it.
